### PR TITLE
Movement classes

### PR DESCRIPTION
**Pull request by NitsanLitov**

### CODE REVIEW

File: MovementBoard/AbstractMovementBoard.cs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    {
+        public Dictionary<ChessColor, (char, int)[,]> MovementBoardByChessColor;
+        public ChessColor[] MovementChessColorByNumber;
+        public ChessColor playerColor;
+        public int maxNumber;
+        public char maxLetter;
+
+        public MovementBoard(ChessColor playerColor, char maxLetter, int maxNumber)
+        {
+            this.playerColor = playerColor;
```
Comment (EranDaniel98) [inline MovementBoard/AbstractMovementBoard.cs:18]: 18 - 20: 
How about using Set and gets? ;)

File: MovementBoard/AbstractMovementBoard.cs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        public List<(char, int)> Left((char, int) currentLocation) { return this.Left(currentLocation, -1); }
+        public List<List<(char, int)>> Diagonal((char, int) currentLocation) { return this.Diagonal(currentLocation, -1); }
+
+        public abstract List<(char, int)> Up((char, int) currentLocation, int maxSteps);
+        public abstract List<(char, int)> Down((char, int) currentLocation, int maxSteps);
+        public abstract List<(char, int)> Right((char, int) currentLocation, int maxSteps);
+        public abstract List<(char, int)> Left((char, int) currentLocation, int maxSteps);
+        public abstract List<List<(char, int)>> Diagonal((char, int) currentLocation, int maxSteps);
+        public abstract List<List<(char, int)>> Knight((char, int) currentLocation);
+        public List<List<(char, int)>> ConvertListToDoubleList(List<(char, int)> locations)
```
Comment (EranDaniel98) [inline MovementBoard/AbstractMovementBoard.cs:39]: Why not protected ? :D

File: MovementBoard/TwoPlayerMovementBoard.cs [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+            locationsList.Add(GetMovementOptions(movementLocation, maxSteps, 1, -1));
+            
+            return locationsList;
+        }
+
+        public override List<List<(char, int)>> Knight((char, int) currentLocation)
+        {
+            (int, int, ChessColor) movementLocation = ConvertToMovementLocation(currentLocation);
+            
+            List<(char, int)> locations = GetMovementOptions(movementLocation, 1, 1, 2);
```
Comment (NitsanLitov) [inline MovementBoard/TwoPlayerMovementBoard.cs:57]: don't use `ConvertListToDoubleList` insert it to a two layered list directly

File: MovementBoard/TwoPlayerMovementBoard.cs [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+        public TwoPlayerMovementBoard(ChessColor playerColor) : base(playerColor, 'h', 8) { }
+
+        public override void SetupMovementBoard()
+        {
+            AddWhiteMovementBoard();
+        }
+
+        public override List<(char, int)> Up((char, int) currentLocation, int maxSteps)
+        {
+            (int, int, ChessColor) movementLocation = ConvertToMovementLocation(currentLocation);
```
Comment (EranDaniel98) [inline MovementBoard/TwoPlayerMovementBoard.cs:19]: why not declare movementLocation in the constructor?
maybe as private - for your judgment

File: MovementBoard/ThreePlayerMovementBoard.cs [hunk 5]
```diff
@@ -0,0 +1,305 @@
...
+        {
+            // Front side
+            (char, int)[,] whiteMovementBoard = new (char, int)[8, 16];
+
+            // First 4 rows
+            for (int number = 1; number <= 4; number++)
+            {
+                for (char letter = 'a'; letter <= 'h'; letter++)
+                {
+                    whiteMovementBoard[number - 1, letter - 'a'] = (letter, number);
```
Comment (EranDaniel98) [inline MovementBoard/ThreePlayerMovementBoard.cs:166]: Change the matrix so that [0,0] is the top left
and [n,n] is the bottom right

**Pull request merged**